explorer/client/src/components/pagination/PaginationWrapper.tsx

```diff
@@ -35,86 +45,162 @@ export default function PaginationWrapper({
-                            disabled={pageIndex === 0}
-                        >
-                            First
-                        </button>
-                    </span>
-
-                    <span className={styles.pagenumber}>
-                        Page {pageIndex + 1} of {FINAL_PAGE_NO}
-                    </span>
-
-                    <span
-                        className={
-                            pageIndex === FINAL_PAGE_NO - 1 ? styles.gone : ''
-                        }
-                    >
-                        <button
-                            id="lastBtn"
-                            disabled={pageIndex === FINAL_PAGE_NO - 1}
-                            onClick={handleLastClick}
-                            className={styles.btncontainer}
-                        >
-                            Last
-                        </button>
-                        <button
-                            id="nextBtn"
-                            className={styles.btncontainer}
-                            disabled={pageIndex === FINAL_PAGE_NO - 1}
-                            onClick={handleNextClick}
-                        >
-                            <svg
-                                width="12"
-                                height="12"
-                                xmlns="http://www.w3.org/2000/svg"
+                                1
+                            </button>
+
+                            <button
+                                className={
+                                    pageIndex === 1
+                                        ? styles.pagenumber
+                                        : styles.btncontainer
+                                }
+                                id="secondBtn"
+                                onClick={handleBtnClick(1)}
+                                disabled={pageIndex === 1}
                             >
-                                <path
-                                    d="M 0 12 L 12 6 L 0 0"
-                                    fill="transparent"
-                                />
-                            </svg>
-                        </button>
-                    </span>
-                </>
-            )}
+                                2
+                            </button>
+                            {pageIndex > 2 && (
+                                <button className={styles.nointeract}>
+                                    ...
+                                </button>
+                            )}
 
-            {viewComponentFn(objectSample)}
+                            {pageIndex > 1 && pageIndex < FINAL_PAGE_NO - 2 && (
+                                <button className={styles.pagenumber}>
+                                    {pageIndex + 1}
+                                </button>
+                            )}
+
+                            {pageIndex < FINAL_PAGE_NO - 3 && (
```

Comment:
![CleanShot 2022-07-19 at 07 23 49](https://user-images.githubusercontent.com/76067158/179774527-eaafa172-544b-46be-8878-33e2ae6eb8b5.png)

It looks like there's still a bug where `...` appears between 2850 and 2851

Reply:
@666lcz that is a different component. I have a few modifications based on the feedback from Dreamten.

Reply:
I see. But is there any reason that we don't consolidate them and use one component for both?

Reply:
I think we can

Reply:
@Andrew47 can we use the same component for both?

I also notice that there's some overflow if the page number is large in the component you are using

http://localhost:3000/addresses/0x813ec1ab1e1797c79d1e1fcbebebc27a1fd21f07

![CleanShot 2022-07-19 at 07 35 53](https://user-images.githubusercontent.com/76067158/179777241-9842825a-1d37-4432-837e-919ab3736edf.png)

Reply:
Here's the pagination conversation thread from Slack for quick reference,
https://mysten-labs.slack.com/archives/C03K869B8M7/p1658158419378789

Reply:
Where were you able to get example numbers in the thousands?

Reply:
Ah - I see. This is the Recent Transactions table in the Home Page. This PR focusses on the Address Results Page (see above description). I can handle this in a separate PR or we could get @Jibz1 to take a look given he has been recently working on the Transaction.

Reply:
@Andrew47 I think we should combine the pagination component. The component on the home page already addresses all this, we just need to include an emit method that fires a function on the parent component. I think you already do this.

Reply:
I will merge the two components following the design specified here: https://www.figma.com/file/OzLaRFzevjxdQAbybWEZk0/Sui-Visualize?node-id=2979%3A59684. This is slightly different from what is both in the Home Page and in this PR.

Reply:
This is covered in the following PR: https://github.com/MystenLabs/sui/pull/3333